arangod/Cluster/ClusterComm.cpp

```diff
@@ -332,13 +326,29 @@ void ClusterComm::cleanup() {
 ////////////////////////////////////////////////////////////////////////////////
 
 void ClusterComm::startBackgroundThread() {
-  _backgroundThread = new ClusterCommThread();
+  _roundRobin = 0;
 
-  if (!_backgroundThread->start()) {
-    LOG_TOPIC(FATAL, Logger::CLUSTER)
-      << "ClusterComm background thread does not work";
-    FATAL_ERROR_EXIT();
-  }
+  for(unsigned loop=0; loop<(TRI_numberProcessors()/8+1); ++loop) {
+    ClusterCommThread * thread = new ClusterCommThread();
+
+    if (thread->start()) {
+      _backgroundThreads.push_back(thread);
+    } else {
+      LOG_TOPIC(FATAL, Logger::CLUSTER)
+        << "ClusterComm background thread does not work";
+      FATAL_ERROR_EXIT();
+    } // else
+  } // for
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief choose next communicator via round robin
+////////////////////////////////////////////////////////////////////////////////
+std::shared_ptr<communicator::Communicator> ClusterComm::communicator() {
+  unsigned index;
+
+  index = (++_roundRobin) % _backgroundThreads.size();
```

Comment:
Please fuse assignment and declaration. No need to have them separate.

Reply:
Typically I would agree with you.  This particular line of code could be confusing so I intentionally limited the scope of activities on line 350.  I feel there is slightly better clarity by keeping the complex math independent of declaration.